benchalerts/benchalerts/integrations/github.py

```diff
@@ -66,8 +66,13 @@ def _encode_jwt(app_id: str, private_key: str) -> str:
         payload = {
             "iss": app_id,
             "iat": datetime.datetime.utcnow() - datetime.timedelta(minutes=1),
-            "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=10),
+            # Expiration time must be no more than 10 minutes, but using exactly 10
+            # minutes may be flaky due to system clock disagreements (see
+            # https://github.com/conbench/conbench/issues/1101). We only need this token
+            # briefly anyway.
+            "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=3),
         }
+        log.debug("Payload to encode for JWT: %s", str(payload))
```

Comment:
Oh, we sign our own token here.

Which entity is the authenticator (the entity that will cryptographically validate this token)?

>  Expiration time must be no more than 10 minutes

Where does this constraint come from?

Terminology: exp is an absolute point in time, a deadline. "10 minutes" is what I typically call the lifetime (life time?) :).

Reply:
Good questions; I just clarified the code comment.